source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -728,6 +730,13 @@ eagerly so that any operations waiting on `Connections <#connection>`_ can retry
 as soon as possible. The pool MUST NOT rely on WaitQueueTimeoutMS to clear
 requests from the WaitQueue.
 
+The clearing method MUST provide the option to close any in-use connections as part
```

Comment:
I am wondering whether we should clarify that only `in-use` connections removal should be limited up to the generation at the moment of clear. To prevent the following scheduling (steps 3-5 happen fast):
1. pool.ready() (gen1)
2. pool.clear(inUser:true) (gen2)
3. pool.ready() (gen2)
4. pool.clear(inUse:false) (gen3)
5. pool.ready() (gen3)
6. prune runs and closes inUse with `gen <= gen2` (the correct behaviour is `gen <= gen1`

cc @patrickfreed

Reply:
Yeah this is a really good point. Maybe add the following line:
>The pool MUST only close in use connections whose generation is less than or equal to the generation of the pool at the moment of the clear (before the increment) that used the `closeInUseConnections` flag.

Unfortunately, I don't think there's a non-racy way to unit test this that I can think of.

Reply:
done